openedx_learning/apps/authoring/collections/api.py

```diff
@@ -48,22 +51,24 @@ def create_collection(
     return collection
 
 
-def get_collection(collection_id: int) -> Collection:
+def get_collection(learning_package_id: int, collection_key: str) -> Collection:
     """
     Get a Collection by ID
     """
-    return Collection.objects.get(id=collection_id)
+    return Collection.objects.get_by_key(learning_package_id, collection_key)
 
 
 def update_collection(
-    collection_id: int,
+    learning_package_id: int,
+    key: str,
+    *,
     title: str | None = None,
     description: str | None = None,
```

Comment:
Sorry for not catching this before. :disappointed: 
I think we should be able to update the key here.
```suggestion
    description: str | None = None,
    new_key: str
```

Reply:
Hmm.. if we allow that, it'll change the [Collection's opaque key](https://github.com/openedx/opaque-keys/pull/335), which will disconnect it from any applied tags.. so I think it should be immutable.

Reply:
You're right! My bad!